### PR TITLE
Tasmota HTTP driver

### PR DESCRIPTION
**Pull request by hectorespert**
Add initial support to [Tasmota](https://tasmota.github.io/docs/) devices with using a http driver.

Relates to https://github.com/reef-pi/reef-pi/issues/1273

Used and tested in a planted tank to control lights using a led controller device and to control temperature using a plug device.

### CODE REVIEW

**Comment by hectorespert** [comment]
@ranjib Could you review it?

File: tasmota/http.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+	}
+	c := http.Client{
+		Timeout: 5 * time.Second,
+	}
+	resp, err := c.Do(req)
+	if err != nil {
+		return false
+	}
+	defer resp.Body.Close()
+	msg, _ := ioutil.ReadAll(resp.Body)
```
Comment (ranjib) [inline tasmota/http.go:68]: should not drop the error, check it and return false if its not nil

File: tasmota/http.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+func (m *httpDriver) PWMChannels() []hal.PWMChannel {
+	return []hal.PWMChannel{m}
+}
+
+func (m *httpDriver) PWMChannel(_ int) (hal.PWMChannel, error) {
+	return m, nil
+}
+
+func (m *httpDriver) LastState() bool {
+	uri := fmt.Sprintf("http://%s/cm?cmnd=Power0", m.address)
```
Comment (ranjib) [inline tasmota/http.go:55]: all these string literals should be declared as constant

File: tasmota/http.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+	msg, _ := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != 200 {
+		return false
+	}
+	var result map[string]interface{}
+	err = json.Unmarshal([]byte(msg), &result)
+	if err != nil {
+		return false
+	}
+	return result["POWER"] == "ON"
```
Comment (ranjib) [inline tasmota/http.go:77]: both POWER and ON could be constants

File: tasmota/http.go [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+}
+
+func (f *factory) GetParameters() []hal.ConfigParameter {
+	return f.parameters
+}
+
+func (f *factory) ValidateParameters(parameters map[string]interface{}) (bool, map[string][]string) {
+	var failures = make(map[string][]string)
+
+	if v, ok := parameters[address]; ok {
```
Comment (ranjib) [inline tasmota/http.go:170]: use switch case for better readabaility instead of if/else

**Review by ranjib — APPROVED** [review]
Some minor style comments. rest looks good. I'm merging this, and assuming you have tested it :-) . thank you so much for the diff. Will be awesome if you send a different PR for the style nits

**Pull request merged**